src/Database/Connection.php

```diff
@@ -503,6 +511,7 @@ public function begin()
      * Commits current transaction.
      *
      * @return bool true on success, false otherwise
+     * @throws \Exception
```

Comment:
It is useful so that you know or your IDE knows that you didn't handle an Exception/Exception type on the action level. Many applications just throw a lot of exceptions without knowing it... They should at least handle them and throw them again if they want this behavior.

Reply:
Those here are all invalid IMO. The throws should only be added as primary level (in phpstorm level 0), that means if this method itself throws the exception, otherwise not.
I have written a sniff for this that can be used to automate this. Everything else is not sniffable and thus not enforceable and only makes things worse/confusing.

Reply:
If one looks in the definition/doc of a method it would be useful to know which Exception is thrown besides IDE that depending on the IDE requires PHPDoc or computes it. Especially on interfaces I would not use the level 0 approach.

Reply:
Everything that is not sniffable will just get outdated and wrong down the method invocation chain.
So no, please lets stick to my suggestion.
Also: Only catchable exceptions make sense to note usually. 500 ones that are not supposed to be caught are useless either way here.

Reply:
Not sure if it will be outdated - but I guess this could be a problem. I see the phpdoc in APIs as a contract that you base your code on - especially if it's not your own code this is important.
The best solution with big impact would be to split checked and unchecked Exceptions (e.g. inheritance) to differentiate between Exceptions one should catch. Still I would document this all way long to the method the user of the api calls.